presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClientModule.java

```diff
@@ -14,21 +14,63 @@
 package io.prestosql.plugin.postgresql;
 
 import com.google.inject.Binder;
-import com.google.inject.Module;
+import com.google.inject.Inject;
+import com.google.inject.Provides;
 import com.google.inject.Scopes;
+import com.google.inject.Singleton;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.prestosql.plugin.base.jmx.MBeanServerModule;
 import io.prestosql.plugin.jdbc.BaseJdbcConfig;
+import io.prestosql.plugin.jdbc.JdbcCatalogName;
 import io.prestosql.plugin.jdbc.JdbcClient;
+import io.prestosql.plugin.jdbc.jmx.StatisticsAwareJdbcClient;
+import io.prestosql.spi.type.TypeManager;
+import org.weakref.jmx.guice.MBeanModule;
 
 import static io.airlift.configuration.ConfigBinder.configBinder;
+import static java.util.Objects.requireNonNull;
+import static org.weakref.jmx.guice.ExportBinder.newExporter;
 
 public class PostgreSqlClientModule
-        implements Module
+        extends AbstractConfigurationAwareModule
 {
+    private final JdbcCatalogName catalogName;
+
+    public PostgreSqlClientModule(JdbcCatalogName catalogName)
+    {
+        this.catalogName = requireNonNull(catalogName, "catalogName is null");
```

Comment:
Now, I need to figure out how to pass catalog name here.  Best would be to get it from guice and pass it `generatedNameOf`